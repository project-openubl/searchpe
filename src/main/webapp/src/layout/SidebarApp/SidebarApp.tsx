import React from "react";
import { NavLink } from "react-router-dom";
import { Nav, NavItem, PageSidebar, NavList } from "@patternfly/react-core";

import { Paths } from "Paths";
import { LayoutTheme } from "../LayoutUtils";

export const SidebarApp: React.FC = () => {
  const renderPageNav = () => {
    return (
      <Nav id="nav-sidebar" aria-label="Nav" theme={LayoutTheme}>
        <NavList>
          <NavItem>
            <NavLink to={Paths.consultaRuc} activeClassName="pf-m-current">
              RUC
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              to={Paths.contribuyenteList}
              activeClassName="pf-m-current"
            >
              Search
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to={Paths.versionList} activeClassName="pf-m-current">
              Versions
            </NavLink>
          </NavItem>
        </NavList>
      </Nav>
    );
  };

  return <PageSidebar nav={renderPageNav()} theme={LayoutTheme} />;
};
