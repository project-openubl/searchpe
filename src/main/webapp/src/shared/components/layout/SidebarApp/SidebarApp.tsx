import React from "react";
import { NavLink } from "react-router-dom";

import { Nav, PageSidebar, NavList } from "@patternfly/react-core";
import { css } from "@patternfly/react-styles";

import { Paths } from "Paths";
import { isAdvancedSearchEnabled, isAuthDisabled, Permission } from "Constants";
import { LayoutTheme } from "../LayoutUtils";
import { VisibilityByPermission } from "shared/containers";

export const SidebarApp: React.FC = () => {
  const renderPageNav = () => {
    return (
      <Nav id="nav-sidebar" aria-label="Nav" theme={LayoutTheme}>
        <NavList title="Consultas">
          <VisibilityByPermission
            hasAny={[Permission.admin, Permission.search]}
          >
            {isAdvancedSearchEnabled() && (
              <NavLink
                to={Paths.contribuyentes}
                className={({ isActive }) =>
                  css("pf-c-nav__link", isActive ? "pf-m-current" : "")
                }
              >
                Buscar
              </NavLink>
            )}
            <NavLink
              to={Paths.consultaRuc}
              className={({ isActive }) =>
                css("pf-c-nav__link", isActive ? "pf-m-current" : "")
              }
            >
              NÃºmero documento
            </NavLink>
          </VisibilityByPermission>
          <VisibilityByPermission
            hasAny={[Permission.admin, Permission.version_write]}
          >
            <NavLink
              to={Paths.versiones}
              className={({ isActive }) =>
                css("pf-c-nav__link", isActive ? "pf-m-current" : "")
              }
            >
              Versiones
            </NavLink>
          </VisibilityByPermission>
          {!isAuthDisabled() && (
            <VisibilityByPermission
              hasAny={[Permission.admin, Permission.user_write]}
            >
              <NavLink
                to={Paths.users}
                className={({ isActive }) =>
                  css("pf-c-nav__link", isActive ? "pf-m-current" : "")
                }
              >
                Usuarios
              </NavLink>
            </VisibilityByPermission>
          )}
        </NavList>
      </Nav>
    );
  };

  return <PageSidebar nav={renderPageNav()} theme={LayoutTheme} />;
};
