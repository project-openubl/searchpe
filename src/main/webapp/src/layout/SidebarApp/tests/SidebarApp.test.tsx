import React from "react";
import { shallow } from "enzyme";
import { SidebarApp } from "../SidebarApp";

it("Test snapshot", () => {
  const wrapper = shallow(<SidebarApp />);
  expect(wrapper).toMatchSnapshot();
});
