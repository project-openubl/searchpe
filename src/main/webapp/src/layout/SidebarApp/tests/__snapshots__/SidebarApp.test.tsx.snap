// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<PageSidebar
  nav={
    <Nav
      aria-label="Nav"
      id="nav-sidebar"
      onSelect={[Function]}
      onToggle={[Function]}
      ouiaSafe={true}
      theme="dark"
    >
      <NavList
        ariaLeftScroll="Scroll left"
        ariaRightScroll="Scroll right"
      >
        <NavItem>
          <NavLink
            activeClassName="pf-m-current"
            to="/consulta-ruc"
          >
            RUC
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            activeClassName="pf-m-current"
            to="/contribuyentes"
          >
            Search
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            activeClassName="pf-m-current"
            to="/versions"
          >
            Versions
          </NavLink>
        </NavItem>
      </NavList>
    </Nav>
  }
  theme="dark"
/>
`;
